pkg/controller/controllerutil/controllerutil.go

```diff
@@ -32,6 +34,37 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 )
 
+const inClusterNamespacePath = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+
+var (
+	// Will store memoized controller namespace
+	inClusterNamespace = make(map[string]string)
+)
+
+// GetInClusterNamespace returns the namespace by which the controller is running under.
+// Use file to pass in a different file. If "", will use the default location,
+// "/var/run/secrets/kubernetes.io/serviceaccount/namespace".
+// Note that the resolved namespace is persisted after it is read from disk.
+func GetInClusterNamespace(file string) (string, error) {
+	if inClusterNamespace, ok := inClusterNamespace[file]; ok {
```

Comment:
Could we start with no arguments? Let's see if folks need this to be configurable, passing an empty string seems a bit odd

Reply:
what's the best way to handle the test of the function?

Reply:
We could make `inClusterNamespacePath` a variable and change it for testing? Or can we write at that path during unit tests?